OrbitGl/TimeGraphLayout.cpp

```diff
@@ -162,3 +167,38 @@ void TimeGraphLayout::SetSortedThreadIds(
 
 //-----------------------------------------------------------------------------
 void TimeGraphLayout::Reset() { m_DrawFileIO = false; }
+
+//-----------------------------------------------------------------------------
+#define FLOAT_SLIDER(x)                     \
+  if (ImGui::SliderFloat(#x, &x, 0, 100)) { \
+    needs_redraw = true;                    \
+  }
+
+//-----------------------------------------------------------------------------
+#define FLOAT_SLIDER_MIN_MAX(x, min, max)     \
+  if (ImGui::SliderFloat(#x, &x, min, max)) { \
+    needs_redraw = true;                      \
+  }
```

Comment:
is there a reason for using macro instead of function?

Reply:
To have access to the preprocessor's "stringify" functionality ("#x"), and to reduce a block of ugly boilerplate code to a short one liner debug utility.

```
FLOAT_SLIDER(m_SpaceBetweenTracksAndThread);
FLOAT_SLIDER(m_SpaceBetweenCoresAndThread);
```
vs
```
  if (ImGui::SliderFloat("m_SpaceBetweenTracksAndThread",
                         &m_SpaceBetweenTracksAndThread, 0, 100.f)) {
    needs_redraw = true;
  }
  if (ImGui::SliderFloat("m_SpaceBetweenCoresAndThread",
                         &m_SpaceBetweenCoresAndThread, 0, 100.f)) {
    needs_redraw = true;
  }
```